src/Plugin/InlineEntityForm/EntityInlineEntityFormController.php

```diff
@@ -357,4 +395,118 @@ protected function buildChildFormState(EntityFormInterface $controller, FormStat
+    }
+
+    $form['actions'] = [
+      '#type' => 'container',
+      '#weight' => 100,
+    ];
+    $form['actions']['ief_' . $form['#op'] . '_save'] = [
+      '#type' => 'submit',
+      '#value' => $save_label,
+      '#name' => 'ief-' . $form['#op'] . '-submit-' . $delta,
+      '#limit_validation_errors' => [$form['#parents']],
+      '#attributes' => ['class' => ['ief-entity-submit']],
+      '#ajax' => [
+        'callback' => 'inline_entity_form_get_element',
+        'wrapper' => 'inline-entity-form-' . $form['#ief_id'],
+      ],
+    ];
+    $form['actions']['ief_' . $form['#op'] . '_cancel'] = [
+      '#type' => 'submit',
+      '#value' => t('Cancel'),
+      '#name' => 'ief-' . $form['#op'] . '-cancel-' . $delta,
+      '#limit_validation_errors' => [],
+      '#ajax' => [
+        'callback' => 'inline_entity_form_get_element',
+        'wrapper' => 'inline-entity-form-' . $form['#ief_id'],
+      ],
+    ];
+  }
+
+  /**
+   * Adds submit handlers to the inline entity form.
+   *
+   * @param array $form
+   *   Form array structure.
+   */
+  protected function addSubmitHandlers(&$form) {
+    if ($form['#op'] == 'add') {
+      $form['actions']['ief_add_save']['#submit'] = [
+        'inline_entity_form_trigger_submit',
+        'inline_entity_form_close_child_forms',
+        'inline_entity_form_close_form',
+      ];
+      $form['actions']['ief_add_cancel']['#submit'] = [
+        'inline_entity_form_close_child_forms',
+        'inline_entity_form_close_form',
+        'inline_entity_form_cleanup_form_state',
+      ];
+    }
+    else {
+      $form['actions']['ief_edit_save']['#ief_row_delta'] = $form['#ief_row_delta'];
+      $form['actions']['ief_edit_cancel']['#ief_row_delta'] = $form['#ief_row_delta'];
+
+      $form['actions']['ief_edit_save']['#submit'] = [
+        'inline_entity_form_trigger_submit',
+        'inline_entity_form_close_child_forms',
+        'inline_entity_form_close_row_form',
+      ];
+      $form['actions']['ief_edit_cancel']['#submit'] = [
+        'inline_entity_form_close_child_forms',
+        'inline_entity_form_close_row_form',
+        'inline_entity_form_cleanup_row_form_state',
+      ];
+    }
+  }
+
+  /**
+   * "Creates" entity that is being edited/created in inline form.
+   *
+   * Entity will either be created (when creating) or loaded form form state
+   * (when editing).
+   *
+   * @param $form
+   *   Form structure array.
+   * @param FormStateInterface $form_state
+   *   Form state object.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface
+   *   Created entity object.
+   */
+  protected function createEntity($form, FormStateInterface $form_state) {
+    if ($form['#op'] == 'add') {
+      // Create a new entity that will be passed to the form.
+      $ief_settings = $form_state->get(['inline_entity_form', $form['#ief_id']]);
+      if (isset($ief_settings['form settings']['bundle'])) {
```

Comment:
You could do a !empty here and then below and then you could use elseif for these ifs